solidity/src/Polymer7683.sol

```diff
@@ -0,0 +1,176 @@
+        emit DestinationContractUpdated(chainId, contractAddress);
+    }
+
+    // ============ External Functions ============
+    /**
+     * @notice Process a settlement proof from a destination chain
+     * @param eventProof The proof of the Fill event from the destination chain
+     * @param eventProof The proof of the Fill event from the destination chain
+     */
+    function handleSettlementWithProof(bytes calldata eventProof) external {
+        // Verify event using Polymer prover
+        (
+            uint32 provenChainId,
+            address emitter,
+            ,  // topics
+            bytes memory data
+        ) = prover.validateEvent(eventProof);
+
+        // Verify destination contract is registered for the proven chain
+        address expectedEmitter = destinationContracts[provenChainId];
+        if (expectedEmitter == address(0)) revert UnregisteredDestinationChain();
+
+        // Validate emitter matches registered destination
+        if (emitter != expectedEmitter) revert InvalidEmitter();
+
+        // Decode data from the Filled event format
+        (bytes32 eventOrderId, bytes memory originData, bytes memory fillerData) = abi.decode(data, (bytes32, bytes, bytes));
+
+        // Process settlement with the message origin, sender, order ID, and receiver
+        _handleSettleOrder(
+            provenChainId,
+            TypeCasts.addressToBytes32(emitter),
+            eventOrderId,
+            abi.decode(fillerData, (bytes32))
+        );
+
+        // _handleSettleOrder checks eligibility with
+        // _checkOrderEligibility and simply returns early without
+        // reverting if the order isn't eligible. we need to check if
+        // the order was settled successfully and revert if not.
+        if (orderStatus[eventOrderId] != SETTLED) revert SettlementFailed();
+    }
+
+    /**
+     * @notice Process a refund proof from a destination chain
+     * @param orderId The order ID being refunded
+     * @param eventProof The proof of the Refund event from the destination chain
+     */
+    function handleRefundWithProof(bytes32 orderId, bytes calldata eventProof) external {
+        // Verify event using Polymer prover
+        (
+            uint32 provenChainId,
+            address emitter,
+            ,  // topics
+            bytes memory data
+        ) = prover.validateEvent(eventProof);
+    
+        // Verify destination contract is registered for the proven chain
+        address expectedEmitter = destinationContracts[provenChainId];
+        if (expectedEmitter == address(0)) revert UnregisteredDestinationChain();
+    
+        // Validate emitter matches registered destination
+        if (emitter != expectedEmitter) revert InvalidEmitter();
+    
+        (bytes32 eventOrderId, address refundReceiver) = abi.decode(data, (bytes32, address));
+        
+        // Validate orderId matches
+        if (eventOrderId != orderId) revert InvalidEventData();
```

Comment:
I think you are using the wrong event here, and the previous version was correct (https://github.com/BootNodeDev/intents-framework/pull/92/commits/ff48ebcbf559df9b52e424a4c82cbd403f3e5539#diff-3eb6fc30808cd07aa187e69ddcb78e5d055315738e52941c29ab203e07a892e9L119-L129)

Reply:
fixed!